### PR TITLE
Include mock dataset for energy dependent morphology

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is to include a dataset that exhibits energy-dependent morphology, to be utilised for the energy dependent estimator. 

A python file is also include with how the dataset is created

### CODE REVIEW

File: estimators/mock_data/make.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+maker_safe_mask = SafeMaskMaker(methods=["offset-max"], offset_max=2.0*u.deg)
+dataset = maker.run(empty, obs)
+dataset = maker_safe_mask.run(dataset, obs)
+
+# Add the model on the dataset and Poission fluctuate
+dataset.models = model
+dataset.fake()
+
+# Save the energy dependent mock dataset
+dataset.write('dataset_energy_dependent.fits', overwrite=True)
```
Comment (AtreyeeS) [inline estimators/mock_data/make.py:105]: I think if you save it as `.fits.gz` it will take much less space

**Comment by maxnoe** [comment]
You should squash your commits, otherwise the uncompressed file will still be in the git history

**Comment by Astro-Kirsty** [comment]
Sounds good @maxnoe, I'll do it on merge

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . This looks good. 

I was wondering whether putting mock datasets in the `estimators` directory is confusing. I suppose it is fine for now. I would suggest to call the directory `mock_datasets` or `mock_DL4` to make sure those are not simulated DL3 files.

Maybe one general comment. Could it be possible to use click in `make.py` so that additional datasets could reuse the same file with different commands.

File: estimators/mock_data/make.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def evaluation_radius(self):
+        """Evaluation radius (`~astropy.coordinates.Angle`)."""
+        return 2 * np.max([self.sigma_1TeV.value, self.sigma_10TeV.value]) * u.deg
+        
+# Create spatial model        
+spatial_model = MyCustomGaussianModel()
+
+# Define energy and geometry for the dataset
+energy_axis = MapAxis.from_edges(np.logspace(-1.0, 2, 10), unit="TeV", name="energy", interp="log")
+geom = WcsGeom.create(skydir=(5.6, 0.2), width=5*u.deg,
```
Comment (registerrier) [inline estimators/mock_data/make.py:69]: maybe reduce a bit the overall size of the dataset to reduce its size? e.g. 4 degrees width should be enough

**Comment by Astro-Kirsty** [comment]
Thanks @registerrier.

For sure it would be possible to use click. What commands would you suggest for the click, because there are many parameters that could be differed!

File: estimators/mock_DL4/make.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+# Create an in-memory observation
+location = observatory_locations["cta_south"]
+obs = Observation.create(
+    pointing=pointing, livetime=livetime, irfs=irfs, location=location
+)
+
+# Make the MapDataset
+empty = MapDataset.create(geom, name="dataset-simu")
+maker = MapDatasetMaker(selection=["exposure", "background", "psf", "edisp"])
+maker_safe_mask = SafeMaskMaker(methods=["offset-max"], offset_max=2.0*u.deg)
```
Comment (registerrier) [inline estimators/mock_DL4/make.py:97]: Such a small offset_max seems way too low for CTA IRFs.

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . I have left a minor comment inline.

Regarding the click interface, the suggestion was simply to make what is currently in main one function and command, e.g. `energy_dependent_morphology`.

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . This looks good!

**Pull request merged**